src/hotspot/share/gc/g1/g1HeapRegion.inline.hpp

```diff
@@ -100,12 +100,12 @@ inline HeapWord* HeapRegion::advance_to_block_containing_addr(const void* addr,
     cur_block = next_block;
     // Because the BOT is precise, we should never step into the next card
     // (i.e. crossing the card boundary).
-    assert(!G1BlockOffsetTablePart::is_crossing_card_boundary(cur_block, (HeapWord*)addr), "must be");
+    assert(!G1BlockOffsetTable::is_crossing_card_boundary(cur_block, (HeapWord*)addr), "must be");
   }
 }
 
 inline HeapWord* HeapRegion::block_start(const void* addr, HeapWord* const pb) const {
```

Comment:
Can we move the assert `assert(addr >= _hr->bottom() && addr < _hr->top(), "invalid address");` here?

Reply:
Moved.